docs/src/main/asciidoc/kubernetes-awareness.adoc

```diff
@@ -1,18 +1,29 @@
 == Kubernetes Ecosystem Awareness
 
-All of the features described earlier in this guide work equally well, regardless of whether your application is running inside
+All features described earlier in this guide work equally well, regardless of whether your application is running inside
 Kubernetes. This is really helpful for development and troubleshooting.
 From a development point of view, this lets you start your Spring Boot application and debug one
 of the modules that is part of this project. You need not deploy it in Kubernetes,
 as the code of the project relies on the
 https://github.com/fabric8io/kubernetes-client[Fabric8 Kubernetes Java client], which is a fluent DSL that can
 communicate by using `http` protocol to the REST API of the Kubernetes Server.
 
-To disable the integration with Kubernetes you can set `spring.cloud.kubernetes.enabled` to `false`. Please be aware that when `spring-cloud-kubernetes-config` is on the classpath,
-`spring.cloud.kubernetes.enabled` should be set in `bootstrap.{properties|yml}` (or the profile specific one),  otherwise it should be in `application.{properties|yml}` (or the profile specific one).
-Because of the way we set up a specific `EnvironmentPostProcessor` in `spring-cloud-kubernetes-config`, you also need to disable that processor via a system property (or an environment variable), for example you could start
-your application via `-DSPRING_CLOUD_KUBERNETES_ENABLED=false` (any form of relaxed binding will work too).
-Also note that these properties: `spring.cloud.kubernetes.config.enabled` and `spring.cloud.kubernetes.secrets.enabled` only take effect when set in `bootstrap.{properties|yml}`
+Kubernetes awareness is based on Spring Boot API, specifically on https://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/autoconfigure/condition/ConditionalOnCloudPlatform.html[ConditionalOnCloudPlatform].
+That property will auto-detect if your application is currently deployed in kubernetes or not. It is possible to override
+that setting via `spring.main.cloud-platform`.
+
+For example, if you need to test some features, but do not want to deploy to a cluster, it is enough to set the:
+`spring.main.cloud-platform=KUBERNETES`. This will make `spring-cloud-kubernetes` act as-if it is deployed in a real cluster.
+Be aware that when `spring-cloud-kubernetes-config` is on the classpath, `spring.main.cloud-platform` should be set in `bootstrap.{properties|yml}`
+(or the profile specific one),  otherwise it should be in `application.{properties|yml}` (or the profile specific one).
+Also note that these properties: `spring.cloud.kubernetes.config.enabled` and `spring.cloud.kubernetes.secrets.enabled`
+only take effect when set in `bootstrap.{properties|yml}`.
+
+=== Functional changes since 2.1.x version
```

Comment:
a light note about the changes, mainly here because I bet we will need to reference it in the future questions/issues that will come, here or on stackoverflow or any other channel.